### DISCUSSION TITLE
Role menu

### DISCUSSION THREAD

**Discussion by FFlyyy** [Pycord-Development/pycord#1201] [Q&A]
Im trying to make a dropdown menu but for roles in pycord (discord.py works too) but I don't know how

Here's an image of what I want it to look like (from the Basement Developers server) 
![unaknown](https://user-images.githubusercontent.com/99058935/160267578-3f471aff-bf52-46f0-9e4c-d6474c41f690.png)

**Comment by clearlakes** [comment]
One way that you could do this would be to set the values of the options in the dropdrown to the IDs of the roles you want (or loop over every role in the guild and add their ID as a value, if you want to list every role); once a selection is made, use the selection's value (the role ID) to get the role and add it to the interaction user.

**Reply by FFlyyy** [reply]
So if I understand I just basically need to make it that in my code I just add `member.add_roles(role ID)` somewhere in my code so where the person clicks the role in the dropdown menu it basically assigns it? Also if you could can you show me where I should add `member.add_roles(role ID)` ?

**Reply by Dorukyum** [reply]
Use the select menu callback.
